books-refractored.js

```diff
@@ -0,0 +1,675 @@
+  previousSort = sortingFunction;
+};
+
+const sortByHighRating = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => b.rating - a.rating);
+  return sortedBooks;
+};
+
+const sortByPublish = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => b.year - a.year);
+  return sortedBooks;
+};
+
+const sortByAuthor = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => {
+    if (a.author < b.author) {
+      return -1;
+    } else if (a.author > b.author) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
+  return sortedBooks;
+};
+
+const sortByTileAZ = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => {
+    const stringA = String(a.title);
+    const stringB = String(b.title);
+
+    if (stringA < stringB) {
+      return -1;
+    } else if (stringA > stringB) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
+  return sortedBooks;
+};
+
+const sortByTileZA = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => {
+    const stringA = String(a.title);
+    const stringB = String(b.title);
+
+    if (stringA < stringB) {
+      return 1;
+    } else if (stringA > stringB) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
+  return sortedBooks;
+};
+
+const getRandomBook = () => {
+  let i;
+  do {
+    i = Math.floor(Math.random() * books.length);
+  } while (i === previousNum); // prevent getting the same number
+  previousNum = i; //update previousNum
+  const randomBook = [books[i]];
+  bookCards(randomBook);
+  console.log(i);
+};
+
+const resetFunc = () => {
+  isGenreFiltered = false;
+  isYearFiltered = false;
+  isSorted = false;
+  previousFilter = null;
+  genreBtns.forEach((genreBtn) => genreBtn.classList.remove("selected"));
+  yearBtns.forEach((yearBtn) => yearBtn.classList.remove("selected"));
+  sortBtns.forEach((sortBtn) => sortBtn.classList.remove("selected"));
+};
+
+/*
+
+
+Add event listener
+
+
+*/
+
+bookDisplay.addEventListener("mouseover", (event) => {
+  const target = event.target;
+  if (target.closest(".book-card")) {
+    target.closest(".book-card").classList.add("book-hover");
+  }
+});
+
+bookDisplay.addEventListener("mouseout", (event) => {
+  const target = event.target;
+  if (target.closest(".book-card")) {
+    target.closest(".book-card").classList.remove("book-hover");
+  }
+});
+
+//
+//----Genre Style
+
+genreBtns.forEach((genreBtn) => {
+  genreBtn.addEventListener("mouseover", () => {
+    if (!genreBtn.classList.contains("selected")) {
+      genreBtn.classList.add("hover");
+    }
+  });
+
+  genreBtn.addEventListener("mouseout", () => {
+    genreBtn.classList.remove("hover");
+  });
+
+  genreBtn.addEventListener("click", () => {
+    genreBtn.classList.remove("hover");
+    toggleGenreSelected(genreBtn);
+  });
+});
```

Comment:
The mouseover and mouseout event listeners on the book cards adds a nice touch. Makes the user experience smoother! Nice! ✨